src/wms/src/main/java/org/geoserver/wms/WMSInfoImpl.java

```diff
@@ -24,6 +24,8 @@ public class WMSInfoImpl extends ServiceInfoImpl implements WMSInfo {
 
     List<String> srs = new ArrayList<>();
 
+    List<String> allowedURLsForAuthForwarding = new ArrayList<>();
```

Comment:
XStream dodges this initialization, leaving the field null, when there is no tag in the XML (so, anyone upgrading).